cloud/aws/bin/deploy.py

```diff
@@ -45,43 +45,59 @@ def _check_application_secret_length(config: ConfigLoader, aws: AwsCli):
-                {Color.END}
+                
+                Additionally, a snapshot will be performed just prior to the upgrade. The snapshot will have a name that starts with "preupgrade". You may also have a snapshot called "{config.app_prefix}-civiform-db-finalsnapshot".{Color.END}
                 '''))
-            if to_apply < current_major:
-                raise ValueError(
-                    f'{Color.RED}Your current version of PostgreSQL appears to be newer than the version specified for this CiviForm release. Ensure you are using the correct version of the cloud-deploy-infra repo and POSTGRESQL_MAJOR_VERSION is unset or set appropriately.{Color.END}'
+            if major_to_apply < current_major:
+                print(
+                    f'{Color.RED}Your current version of PostgreSQL appears to be newer than the version specified for this CiviForm release. Ensure you are using the correct version of the cloud-deploy-infra repo and POSTGRESQL_VERSION is unset or set appropriately.{Color.END}'
                 )
-            if to_apply not in pg_upgrade_table:
-                raise ValueError(
-                    f'{Color.RED}Unsupported upgrade to PostgreSQL version {to_apply} specified for POSTGRESQL_MAJOR_VERSION. If this seems incorrect, contact a CiviForm maintainer.{Color.END}'
+                exit(1)
+            if major_to_apply != 16:
+                print(
+                    f'{Color.RED}Unsupported upgrade to PostgreSQL version {major_to_apply} specified for POSTGRESQL_VERSION. If this seems incorrect, contact a CiviForm maintainer.{Color.END}'
+                )
+                exit(1)
+            if current_major == 12 and current_minor < 17:
+                print(
+                    f'{Color.RED}In order to upgrade to PostgreSQL {major_to_apply}, you must first upgrade to PostgreSQL 12.17 or a later PostgreSQL 12 minor version. You will need to perform this upgrade in the AWS RDS console before proceeding.{Color.END}'
                 )
-            if current_major not in pg_upgrade_table[to_apply]:
+                exit(1)
+            if current_version not in upgrade_path.keys():
                 print(
-                    f'{Color.YELLOW}This version of the deployment tool does not have information about if {current_major}.{current_minor} is sufficiently new enough to upgrade to version {to_apply}. Check https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_UpgradeDBInstance.PostgreSQL.html and verify this is a valid upgrade path.{Color.END}'
+                    f'{Color.YELLOW}This version of the deployment tool does not have information about if {current_version} is sufficiently new enough to upgrade to the latest minor version of version PostgreSQL {major_to_apply}. Check https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_UpgradeDBInstance.PostgreSQL.html and verify this is a valid upgrade path.{Color.END}'
```

Comment:
```suggestion
                    f'{Color.YELLOW}The CiviForm deployment tool does not yet know if {current_version} is sufficiently new enough to upgrade to the latest minor release of PostgreSQL version {major_to_apply}. Check https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_UpgradeDBInstance.PostgreSQL.html and verify this is a valid upgrade path.{Color.END}'
```

This was hard to read with the multiple ambiguous use of the work version. Does this still mean what you wanted?

Reply:
I like your new copy much more. Ship it!